workflow-controller/controller-service/src/main/java/Converters/ServiceSpecConverter.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package Converters;
+
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.constants.ValidationConstants;
+import io.hyscale.commons.models.Activity;
+import io.hyscale.controller.util.ServiceSpecUtil;
+import io.hyscale.servicespec.commons.activity.ServiceSpecActivity;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+
+/*
+Provides parameters and funtions such as profile reference schema,Regex for profile file naming,
+data validation and respective warning and error messages  for service spec validation.
+ */
+public class ServiceSpecConverter extends Converter {
+
+    private static final String SERVICE_SCHEMA_FILE = "/hspec/" + ToolConstants.RELEASE + "/service-spec.json";
+
+    @Override
+    public String getFilePattern() {
+        return ValidationConstants.SERVICE_SPEC_NAME_REGEX;
+    }
+
+    @Override
+    public String getReferenceSchema() {
+        return SERVICE_SCHEMA_FILE;
+    }
+
+    @Override
+    public String getWarnMessage() {
+        return ValidationConstants.INVALID_SERVICE_SPEC_NAME_MSG;
+    }
+
+    @Override
+    public int isDataValid(File serviceSpecFile) throws Exception {
+        String serviceName = ServiceSpecUtil.getServiceNameFromPath(serviceSpecFile.getPath());
+        ServiceSpec serviceSpec = new ServiceSpec(FileUtils.readFileToString(serviceSpecFile, "UTF-8"));
+        return serviceName.equals(serviceSpec.get(HyscaleSpecFields.name).asText()) ? 0 : 1;
+    }
+
```

Comment:
This logic is discussed to be removed right ? No fetching names of service & profile from the file names.